### PR TITLE
Add features for capstone project for module 3 - Formula 1 drivers metrics

### PR DESCRIPTION
**Pull request by ivangonzalez224**
It is my Formula 1 driver metrics project.
It is about displaying a list of metrics that can be filtered out by constructor (brand). The metric is the number of drivers in Formula 1 during the last 10 seasons (2023-2014). When filtering the metrics by any constructor, the details page displays the number of drivers for each season for the specific filtered constructor.
I added the following features:

- Installed the necessary dependencies.
- Added the store and the reducers.
- Added dynamic routing for the whole project.
- Added the components for each page.
- Set up the structure of nav bar and the home and details pages.
- Fetched the metrics data from the API.
- Displayed the list of metrics on the home page.
- Added a search bar component that allows to filter data by constructor name.
- Added styles to the search bar component.
- Added logic to filter metrics by constructor (brand) and update route.
- Rendered the filtered data in the details page.
- Added styles to the nav bar and the home and details pages.
- Added tests for the elements in the nav bar component and the home and details pages.

### CODE REVIEW

File: src/components/home/HomeList.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+          <HomeItem
+            key={driver.id}
+            driverId={driver.id}
+            constructorImg={driver.constructor_img}
+            constructorName={driver.constructor_name.replace('_', ' ')}
+            numDrivers={driver.num_drivers}
+            contIds={driverItems.indexOf(driver) + 1}
+          />
+        ))}
+      </div>
```
Comment (Strangeal) [inline src/components/home/HomeList.js]: - [ ] 👏 Great job so far in building your capstone project! 🚀

    - While your work has been impressive, I'd like to bring your attention to a specific requirement that appears to be missing from the project's specifications.

    - As outlined in the project requirements, there's a crucial element yet to be implemented: the `filter` functionality. This feature is designed to assist users in swiftly sorting through the lists of categories on the home page, focusing particularly on the category of "cars" in your context. Integrating this `filter` functionality holds significance as it enhances user experience by allowing them to save time, avoiding the need to manually scroll through the page in search of a specific car category. Instead, users can simply enter their desired query into the search bar and quickly locate what they're looking for. 🚗💨 See below for more details on project requirements:
    
Screenshot 2023-09-07 at 19 04 16


Please consider prioritizing this feature to ensure your project aligns with the defined requirements and provides a smoother experience for your users. Keep up the excellent work! 🌟👍

**Review by Strangeal — CHANGES_REQUESTED** [review]
Hi @ivangonzalez224, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the features ✅ 
- Nice *UI* design ✅ 
- Professional `Video` Presentation ✅
- Pixel Perfect Design ✅ 
- Acknowledge Design Owner ✅
- Professional `README` template ✅ 
- Correct `GitHub` workflow ✅ 
- Meaningful **Github** commits ✅ 
- Successfully deployed 🚀 ✅ 
- Used `render` for deployment ✅ 
- Correct `.github/workflows` for linters ✅ 
- Followed *best-practices* ✅ 
- Correct **Semantics** used ✅ 
- Well structured project ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/home/HomeList.js (hunk 1 above)
Comment (ivangonzalez224) [reply src/components/home/HomeList.js]: Hi @Strangeal thank you for your feedback.
I already implemented the filter functionality you suggested, by adding a search bar in the home page that allows the user to filter details by entering the constructor name in order to enhance the user experience.

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @ivangonzalez224 ,
# STATUS: APPROVED! 🥇 💯 
done

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**